tools/testing/target_determination/heuristics/filepath.py

```diff
@@ -81,32 +88,56 @@ def file_matches_keyword(file: str, keyword: str) -> bool:
     )
 
 
+def get_freq_dict(tests: list[str], changed_files: list[str]) -> dict[str, int]:
+    keyword_frequency: dict[str, int] = defaultdict(int)
+    for cf in changed_files:
+        keywords = get_keywords(cf)
+        for keyword in keywords:
+            keyword_frequency[keyword] += 1
+
+    test_ratings: dict[str, int] = defaultdict(int)
+
+    for test in tests:
+        for keyword, frequency in keyword_frequency.items():
+            if file_matches_keyword(test, keyword):
+                test_ratings[test] += frequency
+    return test_ratings
+
+
 class Filepath(HeuristicInterface):
     # Heuristic based on folders in the file path.  Takes each folder of each
     # changed file and attempts to find matches based on those folders
     def __init__(self, **kwargs: dict[str, Any]) -> None:
         super().__init__(**kwargs)
 
     def get_prediction_confidence(self, tests: list[str]) -> TestPrioritizations:
-        keyword_frequency: dict[str, int] = defaultdict(int)
         try:
             changed_files = query_changed_files()
         except Exception as e:
             warn(f"Can't query changed test files due to {e}")
             changed_files = []
 
-        for cf in changed_files:
-            keywords = get_keywords(cf)
-            for keyword in keywords:
-                keyword_frequency[keyword] += 1
-
-        test_ratings: dict[str, float] = defaultdict(float)
-
-        for test in tests:
-            for keyword, frequency in keyword_frequency.items():
-                if file_matches_keyword(test, keyword):
-                    test_ratings[test] += frequency
-        test_ratings = {TestRun(k): v for (k, v) in test_ratings.items() if k in tests}
+        test_ratings = get_freq_dict(tests, changed_files)
+        test_ratings = {
+            TestRun(k): float(v) for (k, v) in test_ratings.items() if k in tests
+        }
         return TestPrioritizations(
             tests, normalize_ratings(test_ratings, 0.25, min_value=0.125)
         )
+
+
+if __name__ == "__main__":
+    # Quick thing so you can call the heuristic from the command line with a sha
```

Comment:
This looks like an useful thing to have on other heuristics too (I usually have this on my lambda function to call it locally)